classes/class-wc-connect-settings-pages.php

```diff
@@ -86,13 +86,13 @@ public function output_shipping_settings_screen() {
 			);
 
 			$carriers_response = $this->api_client->get_carrier_accounts();
-			if ( ! is_wp_error( $carriers_response ) && $carriers_response ) {
+			if ( ! is_wp_error( $carriers_response ) && (array) $carriers_response ) {
```

Comment:
Why do we need to cast the response object to an array for a boolean check here?

Reply:
`$this->api_client->get_carrier_accounts()` returns a JSON decoded object representing the server response for this request. Presumably due to the mock, this was somehow returning an empty object, which meant that `( ! is_wp_error( $carriers_response ) && $carriers_response )` evaluated to `true`, because `$carriers_response` wasn't an `WP_Error` and it wasn't empty/falsey. Then the next line, `$carriers_response->carriers`, threw a PHP error notice for undefined index. The array cast here is for that scenario and prevents the error from occurring and consequently that notice from being shown.

Reply:
Instead of `(array) $carriers_response ) `, what do you think if we check the property directly instead? `empty($carriers_response->carriers)`, assuming `$carriers_response` is not `null`. If we want to check `null`, then we would need to do
both `empty($carriers_response)` ` empty($carriers_response->carriers)`

Reply:
Updated this to remove that conniving cast and followed your suggestion!